service/project.go

```diff
@@ -236,13 +232,21 @@ func (uis *UIServer) modifyProject(w http.ResponseWriter, r *http.Request) {
 				return
 			}
 			projectVars.GithubHookID = 0
+			projectRef.TracksPushEvents = false
 		}
 	}
 
 	//modify project vars if necessary
 	projectVars.Vars = responseRef.ProjVarsMap
 	projectVars.PrivateVars = responseRef.PrivateVars
 	projectVars.PatchDefinitions = responseRef.PatchDefinitions
+
+	err = projectRef.Upsert()
+	if err != nil {
+		uis.LoggedError(w, r, http.StatusInternalServerError, err)
+		return
+	}
+
```

Comment:
nit: probably makes more sense to move this bloc before line 239

Reply:
done